src/server.c

```diff
@@ -2654,9 +2654,10 @@ void initServer(void) {
 
     /* Create the Redis databases, and initialize other internal state. */
     int slot_count_bits = (server.cluster_enabled) ? CLUSTER_SLOT_MASK_BITS : 0;
+    int flags = (server.cluster_enabled) ? KVSTORE_ALLOCATE_DICTS_ON_DEMAND | KVSTORE_FREE_EMPTY_DICTS : KVSTORE_ALLOCATE_DICTS_ON_DEMAND;
```

Comment:
```suggestion
    int flags = KVSTORE_ALLOCATE_DICTS_ON_DEMAND;
    if (server.cluster_enabled) flags |= KVSTORE_FREE_EMPTY_DICTS;
```
Minor preference for this, since I think it's clearer to understand.

Reply:
yes, agree, this actually was the way i originally wrote it.